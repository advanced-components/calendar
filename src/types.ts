export interface VEvent {
  title: string;
  start: number;
  end: number;
}
