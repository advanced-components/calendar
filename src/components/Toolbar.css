.cal-toolbar {
  width: 100%;
  border-bottom: 1px solid #eee;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px;
}
