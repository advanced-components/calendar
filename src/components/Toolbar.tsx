import React from 'react';
import styled from 'styled-components';
import { Views, Navigate } from '../types';

export type ToolbarProps = {
  view: Views;
  onNavigate: (to: Navigate) => void;
  onChangeView: (view: Views) => void;
};

export const Toolbar = (props: ToolbarProps) => (
  <StyledToolbar>
    <ButtonGroup>
      <Button
        data-testid='navigate-today'
        onClick={() => props.onNavigate(Navigate.TODAY)}
      >
        Today
      </Button>
      <Button
        data-testid='navigate-previous'
        onClick={() => props.onNavigate(Navigate.PREVIOUS)}
      >
        {'<'}
      </Button>
      <Button
        data-testid='navigate-next'
        onClick={() => props.onNavigate(Navigate.NEXT)}
      >
        {'>'}
      </Button>
    </ButtonGroup>
    <ButtonGroup>
      <Button
        className={props.view === Views.DAY ? 'active' : ''}
        data-testid='change-view-day'
        onClick={() => props.onChangeView(Views.DAY)}
      >
        Day
      </Button>
      <Button
        className={props.view === Views.WEEK ? 'active' : ''}
        data-testid='change-view-week'
        onClick={() => props.onChangeView(Views.WEEK)}
      >
        Week
      </Button>
      <Button
        className={props.view === Views.MONTH ? 'active' : ''}
        data-testid='change-view-month'
        onClick={() => props.onChangeView(Views.MONTH)}
      >
        Month
      </Button>
    </ButtonGroup>
  </StyledToolbar>
);

const StyledToolbar = styled.div`
  grid-column-start: 1;
  grid-column-end: 3;
  width: 100%;
  border-bottom: 1px solid #eee;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 0 16px;
  align-items: center;
`;

const Button = styled.button`
  /* Structure */
  display: inline-block;
  zoom: 1;
  line-height: normal;
  white-space: nowrap;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  user-select: none;
  /* Styles */
  outline: none;
  font-size: 13px;
  padding: 0.5em 1em;
  border: 1px solid rgba(0, 0, 0, 0.2);
  border-radius: 3px;
  text-align: center;
  :hover {
    background-image: linear-gradient(
      transparent,
      rgba(0, 0, 0, 0.05) 40%,
      rgba(0, 0, 0, 0.1)
    );
  }
  &.active {
    background-image: linear-gradient(
      transparent,
      rgba(0, 0, 0, 0.05) 40%,
      rgba(0, 0, 0, 0.1)
    );
  }
`;

const ButtonGroup = styled.div`
  margin: 0;
  border-radius: 0;
  border-right: none;
  ${Button} {
    margin: 0;
    border-radius: 0;
    border-right: none;
  }
  ${Button}:first-child {
    border-top-left-radius: 3px;
    border-bottom-left-radius: 3px;
  }
  ${Button}:last-child {
    border-top-right-radius: 3px;
    border-bottom-right-radius: 3px;
    border-right: 1px solid rgba(0, 0, 0, 0.2);
  }
`;
