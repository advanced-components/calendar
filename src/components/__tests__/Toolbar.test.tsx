import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';
import { Views, Navigate } from '../../types';
import { ToolbarProps, Toolbar } from '../Toolbar';

afterEach(cleanup);

const onNavigate = jest.fn();
const onChangeView = jest.fn();

const defaultProps: ToolbarProps = {
  onNavigate: onNavigate,
  onChangeView: onChangeView
};

describe('Toolbar component', () => {
  it('renders correctly', () => {
    const { container } = render(<Toolbar {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('call onNavigate()', () => {
    const { getByTestId } = render(<Toolbar {...defaultProps} />);
    fireEvent.click(getByTestId('navigate-today'));
    expect(onNavigate).toHaveBeenCalledWith(Navigate.TODAY);
    fireEvent.click(getByTestId('navigate-previous'));
    expect(onNavigate).toHaveBeenCalledWith(Navigate.PREVIOUS);
    fireEvent.click(getByTestId('navigate-next'));
    expect(onNavigate).toHaveBeenCalledWith(Navigate.NEXT);
  });

  it('call onChangeView()', () => {
    const { getByTestId } = render(<Toolbar {...defaultProps} />);
    fireEvent.click(getByTestId('change-view-day'));
    expect(onChangeView).toHaveBeenCalledWith(Views.DAY);
    fireEvent.click(getByTestId('change-view-week'));
    expect(onChangeView).toHaveBeenCalledWith(Views.WEEK);
    fireEvent.click(getByTestId('change-view-month'));
    expect(onChangeView).toHaveBeenCalledWith(Views.MONTH);
  });
});
