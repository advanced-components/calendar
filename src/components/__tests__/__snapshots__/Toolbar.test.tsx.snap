// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar component renders correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa dAlDHg"
  >
    <div
      class="sc-htpNat ilyjRN"
    >
      <button
        class="sc-bwzfXH XSwyB"
        data-testid="navigate-today"
      >
        Today
      </button>
      <button
        class="sc-bwzfXH XSwyB"
        data-testid="navigate-previous"
      >
        &lt;
      </button>
      <button
        class="sc-bwzfXH XSwyB"
        data-testid="navigate-next"
      >
        &gt;
      </button>
    </div>
    <div
      class="sc-htpNat ilyjRN"
    >
      <button
        class="sc-bwzfXH XSwyB"
        data-testid="change-view-day"
      >
        Day
      </button>
      <button
        class="sc-bwzfXH XSwyB"
        data-testid="change-view-week"
      >
        Week
      </button>
      <button
        class="sc-bwzfXH XSwyB"
        data-testid="change-view-month"
      >
        Month
      </button>
    </div>
  </div>
</div>
`;
